packages/core/saved-objects/core-saved-objects-migration-server-internal/src/kibana_migrator_constants.ts

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { IndexTypesMap } from '@kbn/core-saved-objects-base-server-internal';
+
+export enum TypeStatus {
+  Added = 'added',
+  Removed = 'removed',
+  Moved = 'moved',
+  Untouched = 'untouched',
+}
+
+export interface TypeStatusDetails {
+  currentIndex?: string;
+  targetIndex?: string;
+  status: TypeStatus;
+}
+
+// ensure plugins don't try to convert SO namespaceTypes after 8.0.0
+// see https://github.com/elastic/kibana/issues/147344
+export const ALLOWED_CONVERT_VERSION = '8.0.0';
+
+export const DEFAULT_INDEX_TYPES_MAP: IndexTypesMap = {
```

Comment:
Here's the thing, I need a default state for deployments where `.kibana` index doesn't have the `typeIndexMap` information present just yet (as a matter of fact, none of them will, when upgrading to 8.8.0).

This "pre split" map allows having a picture of what the type => index distribution is before the split.
Once we split (e.g. 'cases*' => '.kibana_cases') the logic does not have a way to know what types have been moved, and from where.

We could call it `PRE_SPLIT_INDEX_TYPES_MAP` if you want.

Reply:
> .kibana index doesn't have the typeIndexMap information present just yet

Just trying to understand this better: this is a way to avoid needing to calculate the "current view" from , for example, running an agg query on `.kibana` and `.kibana_task_manager` to get a breakdown of the types in each. Or is there another way to dynamically build the type map for the current view?

Reply:
Let's distinguish between the current breakdown of types (what I referred to as the BEFORE state), and the desired breakdown of types, derived from the `typeRegistry` (the AFTER state).

My proposal stores the current breakdown of types in a centralised way, in the default SO index (aka `.kibana`).
If that information is not present in `.kibana.mapping._meta`, the logic will assume the indices are in the `DEFAULT_INDEX_TYPES_MAP` state.

As discussed in [another thread](https://github.com/elastic/kibana/pull/154151/files#r1158491642), we could consider other approaches:
* Having the information stored in a distributed way.
* Having the information inferred using aggregate queries for **each** SO index (we need a list of them).

Reply:
At some point, I thought that we already have this information stored in each index somehow, cause the `migrationMappingPropertyHashes` already holds a relation of all types present in the index.
I discarded that idea for some reason that I cannot remember 😞 .